Userland/Libraries/LibGUI/VimEditingEngine.cpp

```diff
@@ -1137,9 +1140,17 @@ void VimEditingEngine::switch_to_visual_mode()
 void VimEditingEngine::update_selection_on_cursor_move()
 {
     auto cursor = m_editor->cursor();
-    auto& line = m_editor->current_line();
-    cursor.set_column(min(cursor.column() + 1, line.length()));
-    m_editor->selection()->set_end(cursor);
+    auto start = m_selection_start_position < cursor ? m_selection_start_position : cursor;
+    auto end = m_selection_start_position < cursor ? cursor : m_selection_start_position;
+
+    if (end.column() >= m_editor->current_line().length()) {
+        if (end.line() != m_editor->line_count() - 1)
+            end = { end.line() + 1, 0 };
+    } else {
+        end.set_column(end.column() + 1);
+    }
+
+    m_editor->selection()->set(start, end);
```

Comment:
Was this the original definition before I got rid of `m_selection_start_position`? That still looks a tad bit too complicated. The if statements there might also break `dj`/`dk`, do they still work?

Reply:
It wasn't, but the previous implementation had issue #6014 where the selection range at the end of the line would be invalid.
I counted three cases:
 - Normal case: column + 1
 - End of the line but not last line: wrap to the start of the next line
 - End of the line and last line: do not update the selection since there are no character at the cursor location but the cursor can move there anyway because `in_visual_mode = true`.